src/test/sql/sql_to_ast_translator_test.cpp

```diff
@@ -330,22 +330,30 @@ TEST_F(SQLToASTTranslatorTest, SelectOuterJoin) {
   EXPECT_EQ((*join_node->join_column_ids()).second, ColumnID{0} /* "a" */);
 }
 
-// TODO(mp): Natural Joins are not translated correctly yet. Resolving matching columns is not implemented so far.
-TEST_F(SQLToASTTranslatorTest, DISABLED_SelectNaturalJoin /* #495 */) {
-  const auto query = "SELECT * FROM table_a AS a NATURAL JOIN table_b AS b;";
+TEST_F(SQLToASTTranslatorTest, SelectNaturalJoin) {
+  const auto query = "SELECT * FROM table_a NATURAL JOIN table_c;";
```

Comment:
For example, switching columns in a subquery currently breaks the natural join:

```
(debug)> load src/test/tables/int_float.tbl A
(debug)> load src/test/tables/int_float5.tbl C

(debug)> SELECT * FROM A NATURAL JOIN C;
=== Columns
|       a|       b|       d|
|     int|   float|   float|
=== Chunk 0 ===
|     123|   456.7|    23.7|
===

(debug)> SELECT * FROM A NATURAL JOIN (SELECT d, a FROM C) AS T2;
Exception thrown while compiling query plan:
  No matching columns for natural join found
```

Reply:
I'll look into this.

Reply:
Fixed it. `set_intersection` expected the input to be sorted. Good catch, thank you!

Reply:
I cant think of another example right now, sorry

Reply:
Don't be sorry - I hope there are none. ;)